### PR TITLE
Makes the config module internal

### LINKED ISSUE #874: Make config module internal
The config.py module no longer contains any public APIs. The only publicly visible items are the constants `PROJECT_CONFIG_FN` and `USER_CONFIG_FN`. Should we change this module to `_config`? If so, do we want to keep those two variables public so that users can easily and programmatically see where to expect signac's config files to be stored? To do so I would suggest importing those into `signac/__init__.py` and adding them to `__all__` in that package.

### PR DESCRIPTION
**Pull request by vyasr**
## Description

This PR moves signac.config to signac._config, reflecting that all its members are private. To simplify its usage, all the functions that it contains have had their leading underscores removed since they are now redundant.

## Motivation and Context

Resolves #874 

## Checklist:

- [x] I am familiar with the [Contributing Guidelines](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [Contributor Agreement](https://github.com/glotzerlab/signac/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac/blob/master/contributors.yaml).
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.
- [ ] The [package documentation](https://github.com/glotzerlab/signac/tree/master/doc) and [framework documentation](https://docs.signac.io/) in [signac-docs](https://github.com/glotzerlab/signac-docs) are up to date with these changes.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac/blob/master/changelog.txt) and added any related issue and pull request numbers for future reference.

### CODE REVIEW

File: signac/project.py [hunk 1]
```diff
@@ -20,17 +20,17 @@
 from tempfile import TemporaryDirectory
 from threading import RLock
 
+from ._config import (
+    Config,
+    get_project_config_fn,
+    load_config,
+    locate_config_dir,
+    raise_if_older_schema,
+    read_config_file,
```
Comment (b-butler) [inline signac/project.py:29]: We are now expose these without underscores then to the users though `signac.project`. Is the idea here that most users interact with signac only through the top module?

**Review by b-butler — COMMENTED** [review]
I just have one comment about our import in `signac/project.py`.

File: signac/project.py (hunk 1 above)
Comment (vyasr) [reply signac/project.py:29]: That is a good point, and somewhat related to #876. This issue isn't unique. There are many places where we have modules marked as internal. Should we also require that their contents are always prefixed with an underscore? I don't really mind that, so I can revert https://github.com/glotzerlab/signac/pull/877/commits/f94a2a7f1e77b73f17f3df708b4efad81f2c2c83 if you think so.

Comment (vyasr) [reply signac/project.py]: Reverted.

**Review by b-butler — APPROVED** [review]
:+1:

**Review by bdice — APPROVED** [review]

**Pull request merged**